### DISCUSSION TITLE
Tried the example in PlatformIO, I am getting error: 'class HardwareSerial' has no member named 'setTxBufferSize'

### DISCUSSION THREAD

**Discussion by neliel123** [eModbus/eModbus#325] [General]
Hi,
I tried to include this library on my PlatformIO IDE and tried the RTU example https://github.com/eModbus/eModbus/blob/master/examples/RTU04example/main.cpp

But I am getting the above error. Do you have any suggestions on how to fix this?

**Comment by Miq1** [comment]
Of course. As stated in the docs, eModbus now is based on the newer ESP32 Arduino cores (>= 2.0.x), that completelely have changed the UART handling. Thus the newer UART methods are used, including ``setTxBufferSize``.

Upgrade you core and the message will be gone.

**Comment by makru83** [comment]
I am trying to use the Wifi-RTU_Bridge, and I am facing the exact same Issue.
I just installed PlatformIO with Package eModbus, and configured platformio.ini with the Url of your Docs.

How do I upgrade the Core as you recommended?
Btw.: I am using a esp8266 nodemcuv2, or is a esp32 required now?

**Reply by Miq1** [reply]
The ESP8266 is only able to handle one task. The asynchronous TCP code by me_no_dev kind of substitutes a second to do the TCP communications, but most of eModbus requires two full tasks at least.

Therefore you will want to use an ESP32 with eModbus, the ESP8266 is only supported for the async TCP clients and servers.
